pkg/volume/cinder/attacher.go

```diff
@@ -293,10 +361,14 @@ func (detacher *cinderDiskDetacher) Detach(deviceMountPath string, nodeName type
 	}
 
 	if err = detacher.cinderProvider.DetachDisk(instanceID, volumeID); err != nil {
-		glog.Errorf("Error detaching volume %q: %v", volumeID, err)
+		glog.Errorf("Error detaching volume %q from node %q: %v", volumeID, nodeName, err)
+		return err
+	}
+	if err = detacher.waitDiskDetached(instanceID, volumeID); err != nil {
+		glog.Errorf("Error detaching volume %q from node %q: %v", volumeID, nodeName, err)
```

Comment:
A better error msg could be "Error waiting for volume %q to detach from node %q: %v"